pkg/executor/simple.go

```diff
@@ -1924,6 +1977,14 @@ func (e *SimpleExec) executeAlterUser(ctx context.Context, s *ast.AlterUserStmt)
 			fields = append(fields, alterField{"password_lifetime=%?", plOptions.passwordLifetime})
 		}
 
+		if userResource.maxUserConnections >= 0 {
```

Comment:
at L1815 it handles privileges, but it has bug https://github.com/pingcap/tidb/issues/59677, maybe we can unify the privilege check

Reply:
Get it~. It will fix the bug #59677 at complete logic.